src/adapters/scene_direction.js

```diff
@@ -65,10 +65,15 @@ export default class SceneDirection {
   setMainRoute(routeId) {
     let mainRoute = null;
     this.routes.forEach((route) => {
-      if (route.id === routeId) {
+      const isActive = route.id === routeId
+      if (isActive) {
         mainRoute = route
       }
-      this.map.setFeatureState({source: `source_${route.id}`, id: 1}, {isActive: route.id === routeId})
+      this.map.setFeatureState({source: `source_${route.id}`, id: 1}, {isActive})
+
+      try {
+        this.map.setLayoutProperty(`route_${route.id}`, 'icon-image', isActive ? 'walking_bullet_active' : 'walking_bullet_inactive')
+      } catch (e) {}
```

Comment:
I am not fond of this solution. It may cause errors to stay unnoticed in the future.  
Wouldn't it work to simply check `this.vehicle` value before calling `setLayoutProperty` ?

Reply:
I replaced the `try { ... } catch(e) {}` with a simple condition and it works just fine